### ISSUE TITLE
[RFE] Helm Chart support for Kubernetes 1.27 - Kustomize - Error: Error: chart requires kubeVersion: < 1.26.0-0 which is incompatible with Kubernetes v1.27.0. Even breaks on Kubernetes 1.25

### ISSUE THREAD

**Issue by HariSekhon** [rancher/rancher#41911]
**Is your feature request related to a problem? Please describe.**

The rancher server helm chart breaks with a version dependency when trying to install on Kubernetes 1.27:

```
Error: Error: chart requires kubeVersion: < 1.26.0-0 which is incompatible with Kubernetes v1.27.0
```

https://github.com/HariSekhon/Kubernetes-configs/actions/runs/5306667657/jobs/9604480991#step:12:901

My full config is here:

https://github.com/HariSekhon/Kubernetes-configs/tree/master/rancher/base

I have deployed this before on EKS on an older version of Kubernetes.

You can reproduce this from the git checkout via `kustomize build --enable-helm`.

What's odd is that I'm hitting this issue even on a machine with kubectl that should be within version:

```shell
$ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.25.4
Kustomize Version: v4.5.7
Server Version: v1.25.4
```
```shell
$ kustomize build --enable-helm
Error: Error: chart requires kubeVersion: < 1.26.0-0 which is incompatible with Kubernetes v1.26.0
```
This error looks like another bug, because `kubectl version` clearly shows using version 1.25 for both `kubectl` and running kubernetes server version...

**Describe the solution you'd like**

Helm chart should be installable on Kubernetes 1.27 clusters

**Describe alternatives you've considered**

I'd already disabled PSP in the helm chart values.yaml:

https://github.com/HariSekhon/Kubernetes-configs/blob/714cfbdb57a789c8cc591509b124b07b2d4fedb5/rancher/base/values.yaml#L32

as I believed that would have generated a yaml that wouldn't work on Kubernetes 1.27

But it looks like the chart itself has an explicit version dependency that won't allow it to even try to install, as per the error message at the top of this ticket?

**Comment by HariSekhon** [comment]
Still relevant

**Comment by brandond** [comment]
As documented at https://github.com/kubernetes-sigs/kustomize/blob/master/examples/chart.md, kustomize runs `helm template` internally. I suspect that the version of helm you have installed is defaulting to Kubernetes 1.27, as kustomize does not appear to pass it the `--kube-version` argument to actually tell the chart what version of Kubernetes you'll be installing the chart to.

I do not think this is an issue with the chart or Rancher itself.

**Comment by HariSekhon** [comment]
@brandond the error implies that the chart will only work with Kubernetes versions < 1.26.0-0 so won't work on Kubernetes 1.27 onwards

I don't think this is as simple as passing the `--kube-version` because the output would still need to be able to run on Kubernetes 1.27+

From what I can tell this is a problem with the chart's version compatibility, probably due to some API object references I expect

**Comment by brandond** [comment]
The chart version restricts you to versions of Kubernetes that Rancher supports. When Rancher supports newer versions of Kubernetes, then the chart is updated. This is not a bug in the chart.

**Comment by zs-dima** [comment]
Latest version works well:
```shell
helm repo add rancher-latest https://releases.rancher.com/server-charts/latest && helm repo update
envsubst < rancher-values.yaml | helm install -n cattle-system rancher rancher-latest/rancher \
  --version 2.8.0-rc3 \
  -f -
```

**Comment by snasovich** [comment]
K8s 1.27 support is available in v2.8.0 per https://github.com/rancher/rancher/issues/41840, closing this issue as duplicate. Note that v2.8.0 is not marked as stable yet but since the reference issue is closed this one is also closed.
